packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -57,20 +56,17 @@ export function updateSyntheticShadowAttributes(vm: VM, template: Template) {
 
     // Apply the new template styling token to the host element, if the new template has any
     // associated stylesheets.
-    if (
-        !isUndefined(newStylesheetTokens) &&
-        !isUndefined(newStylesheets) &&
-        newStylesheets.length !== 0
-    ) {
-        newHostAttribute = newStylesheetTokens.hostAttribute;
-        newShadowAttribute = newStylesheetTokens.shadowAttribute;
-
-        renderer.setAttribute(elm, newHostAttribute, '');
+    if (!isUndefined(newStylesheets) && newStylesheets.length !== 0 && hasShadow(vm)) {
+        newTokens = newStylesheetTokens;
+    }
+
+    if (newTokens) {
```

Comment:
To avoid the type coersion.

```suggestion
    if (!isUndefined(newTokens)) {
```

It is quite that `stylesheetTokens` is defined as optional on the `Template` type. Looking at the compiler the `stylesheetTokens` is [always set](https://github.com/salesforce/lwc/blob/master/packages/@lwc/compiler/src/transformers/template.ts#L72-L75) at by the compiler. This is also true for the `stylesheet` property. I am wondering if we can make both properties required.

Reply:
I'll look into making it required in a separate PR.

Reply:
It looks like it's not possible to make it required, because of this code:

https://github.com/salesforce/lwc/blob/ff6c28bc001b55cdee2a6f9442ce5e95a4f0e96a/packages/%40lwc/engine-core/src/framework/secure-template.ts#L11-L14

I don't understand why we're using a `function` returning an array here as the "default empty template."

Reply:
Went ahead and added a test to this PR.

Reply:
It's certainly something we can reevaluate in a later PR. I haven't looked at the style injection part of the engine holistically for a while now. I am suspecting there are small improvements we can do here.